kind/actions/cluster_actions.go

```diff
@@ -274,28 +274,63 @@ func SetNodeRef(clusterName, nodeName string) error {
 	return nil
 }
 
-func RemoveNode(clusterName, nodeName string) error {
+func GetNodeRefUID(clusterName, nodeName string) (string, error) {
+	// 	k get nodes my-cluster-worker -o custom-columns=UID:.metadata.uid --no-headers
 	allNodes, err := nodes.List(fmt.Sprintf("label=%s=%s", constants.ClusterLabelKey, clusterName))
 	if err != nil {
-		return nil
+		return "", err
 	}
 
 	node, err := nodes.BootstrapControlPlaneNode(allNodes)
 	if err != nil {
-		return err
+		return "", err
 	}
+
+	patch := fmt.Sprintf(`{"spec": {"providerID": "docker://%s"}}`, nodeName)
+	fmt.Println("trying to apply:", patch)
 	cmd := node.Command(
 		"kubectl",
 		"--kubeconfig", "/etc/kubernetes/admin.conf",
-		"delete",
+		"get",
 		"node", nodeName,
+		"--output=custom-columns=UID:.metadata.uid",
+		"--no-headers",
+	)
+	lines, err := exec.CombinedOutputLines(cmd)
+	if err != nil {
+		for _, line := range lines {
+			fmt.Println(line)
+		}
+		return "", errors.Wrap(err, "failed get node ref UID")
+	}
+	return strings.TrimSpace(lines[0]), nil
+}
+
+func DeleteClusterNode(clusterName, nodeName string) error {
+	// get all control plane nodes
+	allNodes, err := nodes.List(fmt.Sprintf("label=%s=%s", constants.ClusterLabelKey, clusterName))
+	if err != nil {
+		return err
+	}
+	var node nodes.Node
+	// pick one that doesn't match the node name we are trying to delete
+	for _, n := range allNodes {
+		if n.Name() != nodeName {
+			node = n
+			break
+		}
+	}
+	cmd := node.Command(
+		"kubectl",
+		"--kubeconfig", "/etc/kubernetes/admin.conf",
+		"delete", "node", nodeName,
 	)
 	lines, err := exec.CombinedOutputLines(cmd)
 	if err != nil {
 		for _, line := range lines {
 			fmt.Println(line)
 		}
-		return errors.Wrap(err, "failed to remove node from cluster")
+		return errors.Wrap(err, "failed update providerID")
```

Comment:
I think the correct message is "failed to remove node from cluster" as this code is in `DeleteClusterNode` function

Reply:
very correct, good eyes